### PR TITLE
Simplify the KeyBindings module and its translations

### PR DESCRIPTION
**Pull request by jsnajdr**
This patch simplifies the `KeyBindings` module and converts it into a simple export of a constant static data structure.

The `KeyBindings` object is used to initialize the global keyboard shortcuts (that's why it's loaded on Calypso boot) and also to populate the keyboard shortcuts modal that appears after pressing 'Shift-/'.

The description strings are no longer statically translated when evaluating the module, but instead we're exporting functions in the form:
```js
translate => translate( 'Shortcut Description' )
```
as the `description.text` fields. This allows to translate the strings using the standard `localize` HOC. The translations are no longer done on Calypso boot, but instead only when the modal is displayed. That removes a few milliseconds from boot time, but also simplifies the module: for example, we no longer need to listen for the `change` event from the `i18n` object and fire `language-change`
events from `KeyBindings`. Also, the return value from `KeyBindings.get` is not re-created on each call, but it's a static object directly exported.

I also removed the test for `key-bindings.js`, because it became tautological -- just testing a constant static export.

**How to test:**
- check that all keyboard shortcuts still work as before
- check that the modal with shortcuts description appears after 'Shift-/'. Beware that this feature is turned off on staging and production, and only enabled in development and wpcalypso/horizon.

### CODE REVIEW

File: client/lib/keyboard-shortcuts/key-bindings.js [hunk 1]
```diff
@@ -1,201 +1,174 @@
 /** @format */
 
-/**
- * External dependencies
- */
+// Returns a function that returns a translated text when applied
+const translate = text => tr => tr( text );
```
Comment (lamosty) [inline client/lib/keyboard-shortcuts/key-bindings.js]: Can we rename it somehow? People might confuse it with the `translate` from `i18n-calypso`.

Comment (jsnajdr) [reply client/lib/keyboard-shortcuts/key-bindings.js]: You mean the `translate` function? I think it's actually very important that it has exactly this name -- there is a l10n script that searches the sources for instances of `translate( 'something' )` and generates the list of strings to translate. @akirk or @yoavf can confirm if that's correct or not.

Comment (akirk) [reply client/lib/keyboard-shortcuts/key-bindings.js]: Correct, we explicitly look for the function call named `translate`, see https://github.com/Automattic/i18n-calypso/blob/master/cli/index.js#L20 resp. https://github.com/Automattic/xgettext-js/blob/master/xgettext.js#L214

File: client/lib/keyboard-shortcuts/index.js [hunk 2]
```diff
@@ -12,18 +14,12 @@ const keymaster = typeof window === 'undefined' ? undefined : require( 'keymaste
...
+import KEY_BINDINGS from './key-bindings';
 
-// flatten the key-bindings object to create an array of key-bindings
-Object.keys( keyBindings ).forEach( function( category ) {
-	flatKeyBindings = flatKeyBindings.concat( keyBindings[ category ] );
-} );
+// Flatten the key-bindings object to create an array of key-bindings. `_.flatMap` converts
+// object in shape `{ c1: [ k1, k2 ], c2: [ k3, k4 ] }` to `[ k1, k2, k3, k4 ]`, i.e., flat-maps
+// over the object property values.
+const flatKeyBindings = flatMap( KEY_BINDINGS );
```
Comment (ockham) [inline client/lib/keyboard-shortcuts/index.js]: :heart:

Comment (jsnajdr) [reply client/lib/keyboard-shortcuts/index.js]: Oh yeah, Lodash can be really cool sometimes 🥇

**Pull request merged**